core/src/main/scala/io/finch/Endpoint.scala

```diff
@@ -129,6 +129,13 @@ trait Endpoint[A] { self =>
     override def toString = self.toString
   }
 
+  def flatMap[B](fn: A => Endpoint[B]): Endpoint[B] = new Endpoint[B] {
+    override def apply(input: Input): Option[(Input, () => Future[Output[B]])] = for {
+      (r1, ao) <- self(input)
+      (r2, bo) <- fn(Await.result(ao()).value)(r1)
```

Comment:
I'd be happy to accept this as long as there in `Await.result` in the code (except for tests). If you feel adventurous, you might think about changing the endpoint output type to something that'd allow you to write flatMap with awaits. Perpahs, #453 might help there in some way.